wgpu_shadertoy/api.py

```diff
@@ -0,0 +1,111 @@
+import json
+import os
+
+import numpy as np
+import requests
+from PIL import Image
+
+
+# TODO: write function that gives a good error message
+def get_api_key():
+    key = os.environ.get("SHADERTOY_KEY", None)
+    if key is None:
+        raise Exception(
+            "SHADERTOY_KEY environment variable not set, please set it to your Shadertoy API key to use API features. Follow the instructions at https://www.shadertoy.com/howto#q2"
+        )
+    return key
+
+
+API_KEY = get_api_key()
+
+HEADERS = {"user-agent": "https://github.com/pygfx/shadertoy script"}
+
+
+def get_shadertoy_by_id(shader_id) -> dict:
+    """
+    Fetches a shader from Shadertoy.com by its ID and returns the JSON data as dict.
+    """
+    url = f"https://www.shadertoy.com/api/v1/shaders/{shader_id}"
+    response = requests.get(url, params={"key": API_KEY}, headers=HEADERS)
+    if response.status_code != 200:
+        raise Exception(
+            f"Failed to load shader at https://www.shadertoy.com/view/{shader_id}"
+        )
+    if "Error" in response.json():
+        raise Exception(
+            f"Shadertoy API error: {response.json()['Error']}, perhaps the shader isn't set to `public+api`"
+        )
+    return response.json()
+
+
+# TODO: consider caching media locally?
+def download_media_channels(inputs):
+    from . import ShadertoyChannel  # lazy import to avoid circular imports?
```

Comment:
If you can, move ShadertoyChannel into it's own file. That's a better way to resolve circular import problems.

Reply:
this whole class requires a refactor anyways once we add more input options than just textures, especially buffers.